lib/galaxy/tools/__init__.py

```diff
@@ -622,6 +622,21 @@ def parse(self, tool_source, guid=None):
 
         self.parse_command(tool_source)
         self.environment_variables = self.parse_environment_variables(tool_source)
+        self.tmp_directories = tool_source.parse_tmp_directories()
+
+        home_target = tool_source.parse_home_target()
+        tmp_target = tool_source.parse_tmp_target()
+        # If a tool explicitly sets one of these variables just respect that and turn off
+        # explicit processing by Galaxy.
+        for environment_variable in self.environment_variables:
+            if environment_variable.get("name") == "HOME":
+                home_target = None
```

Comment:
If it is HOME you can `continue` I guess.

Reply:
+1. It also seems weird that tools should be able to set these vars but I guess this a feature for local tool devs or Galaxies?

Reply:
There was a CWL tool in the GA4GH that set this I believe. It is not so common in the GalaxyXML/BioConda world - but if you take a container-first approach to development I guess one can see why someone might want to do this. If we want to disallow this edge case we maybe could - but the backend should support it - just not the Galaxy tool parser IMO.

I'll rebase with the continue and break requested.

Reply:
No, I think this is ok, just wanted to make sure I understood what was happening. I appreciate the explanation.